e2e/src/helper/MasterAccountManager.ts

```diff
@@ -0,0 +1,405 @@
+ * Automatically detects if using local network by checking if NETWORK env var contains "local"
+ * @returns The appropriate master account (PrivateKeyAccount)
+ */
+export const getMasterAccount = (): PrivateKeyAccount => {
+  const network = process.env['NETWORK'];
+  const isLocal = network?.includes('local');
+
+  if (isLocal) {
+    return privateKeyToAccount(
+      process.env['LOCAL_MASTER_ACCOUNT'] as `0x${string}`,
+      { nonceManager }
+    );
+  } else {
+    return privateKeyToAccount(
+      process.env['LIVE_MASTER_ACCOUNT'] as `0x${string}`,
+      { nonceManager }
+    );
+  }
+};
+
+type MasterAccountStatus = {
+  privateKey: `0x${string}`;
+  address: string;
+  balance: string;
+  available: boolean;
+}
+
+type MasterAccountPool = MasterAccountStatus[];
+
+export const getMasterAccountFromPool = (): PrivateKeyAccount => {
+  const network = process.env['NETWORK'];
+  const isLocal = network?.includes('local');
+
+  if (isLocal) {
+    throw new Error("🚨 Local network not supported for master account from pool");
+  } else {
+
+    // read from the master-account-pool.json file
+    const masterAccountPool: MasterAccountPool = JSON.parse(fs.readFileSync('master-account-pool.json', 'utf8'));
+
+    // Find and return the first available master account
+    const availableMasterAccount: MasterAccountStatus = masterAccountPool.find(account => account.available);
+    if (!availableMasterAccount) {
+      throw new Error("🚨 No available master account found in the pool");
+    }
+
+    return privateKeyToAccount(availableMasterAccount.privateKey);
+  }
+};
+
+export const getTotalMasterAccountsInPool = (): number => {
+  const network = process.env['NETWORK'];
+  const isLocal = network?.includes('local');
+
+  if (isLocal) {
+    throw new Error("🚨 Local network not supported for master account from pool");
+  } else {
+    const masterAccountPool: MasterAccountPool = JSON.parse(fs.readFileSync('master-account-pool.json', 'utf8'));
+    return masterAccountPool.length;
+  }
+}
+
+export const getMasterAccountsFromPool = (): MasterAccountPool => {
+  const network = process.env['NETWORK'];
+  const isLocal = network?.includes('local');
+
+  if (isLocal) {
+    throw new Error("🚨 Local network not supported for master account from pool");
+  } else {
+    let masterAccountPool: MasterAccountPool = JSON.parse(fs.readFileSync('master-account-pool.json', 'utf8'));
+
+    console.log("...Deleting empty entries from master account pool");
+    // Delete any entries that are 
+    // empty 'address'
+    // empty 'balance'
+    // empty 'privateKey'
+    masterAccountPool = masterAccountPool.filter(account => account.address && account.balance && account.privateKey);
+
+    // write the updated masterAccountPool to the file
+    fs.writeFileSync('master-account-pool.json', JSON.stringify(masterAccountPool, null, 2));
+
+    return masterAccountPool;
+  }
+}
+
+export const checkAndTopupWalletsFromPool = async ({
```

Comment:
do we have a check here that the master account we are using to disperse funds has enough funds to cover all the other master accounts that need to be topped up.